tracer/src/Datadog.Trace/Configuration/ServiceNames.cs

```diff
@@ -13,9 +13,13 @@ internal class ServiceNames
     {
         private readonly object _lock = new object();
         private Dictionary<string, string> _mappings = null;
+        private bool _unifyServiceNames;
 
-        public ServiceNames(IDictionary<string, string> mappings)
+        public ServiceNames(IDictionary<string, string> mappings, string metadataSchemaVersion)
```

Comment:
Unrelated but just noticed we use the same mutable `ServiceNames` object inside `ImmutableTracerSettings`. I wonder if we shouldn't 🤔

Reply:
> One important difference which hasn't been called out yet seems to be the fact that these "external" spans were previously all implicitly measured (due to being service-entry spans). So they automatically had metrics generated. After this change, they will no longer get any metrics. That seems like it _could_ be an issue - is that something we've thought about? Should we ensure we mark them as measure?

Good callout, and we 100% have to set them as measured before the feature is usable. I was thinking about doing this in a separate PR, but didn't include this anywhere in the PR description. I have two followup PR's that update the schema and then that could help with testing that we set the measured metric.

Can we do this work in a follow up PR and I'll document this decision in the PR description?

Reply:
Yeah I agree this is not ideal. It would be good to fix this in a separate PR

Reply:
This is being addressed in https://github.com/DataDog/dd-trace-dotnet/pull/4043